pkg/providers/docker/docker.go

```diff
@@ -52,6 +55,12 @@ type provider struct {
 	templateBuilder       *DockerTemplateBuilder
 }
 
+// ProviderKubeconfigWriter reads the kubeconfig secret on a docker cluster and copies the contents to a writer.
+type ProviderKubeconfigWriter struct {
```

Comment:
```suggestion
type KubeconfigWriter struct {
```

The package name-spacing already adds this context